docs/adrs/0002-public-endpoints.md

```diff
@@ -0,0 +1,35 @@
+# Public endpoints for callbacks and docs
+
+Date: 2023-09-14
+
+## Status
+
+Accepted
+
+## Context
+
+To prevent misuse, access to the API has been restricted using Aptible IP filtering. This does not work
+for callbacks from providers as the IPs may change for Twilio and Mailgun. Also, it is desirable
+to have the API docs be publicly available without restriction.
+
+Callback endpoints are secured using signature verification with a shared secret from the provider.
+
+Note: CfA has purchased Twilio Security Edition that supports a static list of IPs from which callbacks originate. As we had already implemented
+signature verification, we have not availed ourselves of this feature. We do not have a similar facility with Mailgun.
+
+## Decision
+
+We will listen on a separate port for callbacks and docs. This will not be secured with IP filtering.
+The specific port can be chosen using the configuration variable `server.trustedPort`. The callbacks and docs will be
+made available under the path `/public/`.
+
+| Port | Use | Security | Path |
+|-|-|-|-|
+| 8080 | Messaging API | IP Filtering & BasicAuth shared password | https://app-54372.on-aptible.com |
+| [8086](/src/main/resources/application.properties) | Messaging Provider Callbacks & API Documentation | Public w/ Provider Signature varification | https://staging.messaging.cfa-platforms.org/public |
```

Comment:
@vrajmohan I added this table to make the info easier to see at a glance. I know it is subject to change, especially the port, but it seems unlikely to me that it will.

Reply:
IMO, the ADR does not need implementation details like a README or INSTALLING. I expect the engineers to read the code to get those details. I would rather focus the ADR on pros and cons of the decisions made.

Reply:
I added the table to make it easier to visually process the information, that there are two ports with different security measures and different urls. The actual ports and urls are not relevant, but I think seeing them broken down in a table helps it click. I agree that this should not be the "source" for this info, and maybe it is risky to have it shown here, but I think it helps clarify the impact of this ADR.

Reply:
Updated the table to remove implementation details